models/gif_psc_exp.cpp

```diff
@@ -0,0 +1,469 @@
+{
+  // use standard names whereever you can for consistency!
+  insert_( names::V_m, &gif_psc_exp::get_V_m_ );
+  insert_( names::E_sfa, &gif_psc_exp::get_E_sfa_ );
+  insert_( names::input_currents_ex, &gif_psc_exp::get_input_currents_ex_ );
+  insert_( names::input_currents_in, &gif_psc_exp::get_input_currents_in_ );
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::gif_psc_exp::Parameters_::Parameters_()
+  : g_L_( 4.0 )         // nS
+  , E_L_( -70.0 )       // mV
+  , c_m_( 80.0 )        // pF
+  , V_reset_( -55.0 )   // mV
+  , delta_u_( 1.5 )     // mV
+  , v_t_star_( -35 )    // mV
+  , lambda0_( 10000.0 ) // Hz
+  , I_e_( 0.0 )         // pA
+  , t_ref_( 4.0 )       // ms
+  , tau_ex_( 2.0 )      // in ms
+  , tau_in_( 2.0 )      // in ms
+{
+  tau_sfa_.clear();
+  q_sfa_.clear();
+  tau_stc_.clear();
+  q_stc_.clear();
+}
+
+nest::gif_psc_exp::State_::State_()
+  : y0_( 0.0 )
+  , y3_( -70.0 )
+  , q_( 0.0 )
+  , r_ref_( 0 )
+  , i_syn_ex_( 0.0 )
+  , i_syn_in_( 0.0 )
+  , initialized_( false )
+  , add_stc_sfa_( false )
+{
+  q_sfa_elems_.clear();
+  q_stc_elems_.clear();
+}
+
+/* ----------------------------------------------------------------
+ * Parameter and state extractions and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::gif_psc_exp::Parameters_::get( DictionaryDatum& d ) const
+{
+  def< double >( d, names::I_e, I_e_ );
+  def< double >( d, names::E_L, E_L_ );
+  def< double >( d, names::g_L, g_L_ );
+  def< double >( d, names::C_m, c_m_ );
+  def< double >( d, names::V_reset, V_reset_ );
+  def< double >( d, names::delta_u, delta_u_ );
+  def< double >( d, names::v_t_star, v_t_star_ );
+  def< double >( d, "lambda0", lambda0_ );
+  def< double >( d, names::t_ref, t_ref_ );
+  def< double >( d, names::tau_syn_ex, tau_ex_ );
+  def< double >( d, names::tau_syn_in, tau_in_ );
+
+  ArrayDatum tau_sfa_list_ad( tau_sfa_ );
+  def< ArrayDatum >( d, names::tau_sfa, tau_sfa_list_ad );
+
+  ArrayDatum q_sfa_list_ad( q_sfa_ );
+  def< ArrayDatum >( d, names::q_sfa, q_sfa_list_ad );
+
+  ArrayDatum tau_stc_list_ad( tau_stc_ );
+  def< ArrayDatum >( d, names::tau_stc, tau_stc_list_ad );
+
+  ArrayDatum q_stc_list_ad( q_stc_ );
+  def< ArrayDatum >( d, names::q_stc, q_stc_list_ad );
+}
+
+void
+nest::gif_psc_exp::Parameters_::set( const DictionaryDatum& d )
+{
+
+  updateValue< double >( d, names::I_e, I_e_ );
+  updateValue< double >( d, names::E_L, E_L_ );
+  updateValue< double >( d, names::g_L, g_L_ );
+  updateValue< double >( d, names::C_m, c_m_ );
+  updateValue< double >( d, names::V_reset, V_reset_ );
+  updateValue< double >( d, names::delta_u, delta_u_ );
+  updateValue< double >( d, names::v_t_star, v_t_star_ );
+  updateValue< double >( d, "lambda0", lambda0_ );
```

Comment:
You should probably convert lambda0 from 1/s to 1/ms (i.e., divide by 1000) here, then you do not need to do that on every time step. You need to multiply by 1000 then in the `get()` method. I suppose it would also make sense to test that lambda0 >= 0 (== 0 means no spiking, but it causes no problems as far as I can see, so one could allow it for testing).